packages/client/hmi-client/src/page/data-explorer/components/tera-selected-resources-header-pane.vue

```diff
@@ -52,22 +54,37 @@ const addResourcesToProject = async (projectId: string) => {
 	// send selected items to the store
 	props.selectedSearchItems.forEach(async (selectedItem) => {
 		if (isDocument(selectedItem)) {
-			const body: ExternalPublication = {
-				xdd_uri: (selectedItem as Document).gddId,
-				title: (selectedItem as Document).title
+			const document = selectedItem as Document;
+			const name = document.title;
+
+			// get the project username
+			const project = await getProject(projectId);
+			const username = project?.username ?? '';
+
+			// get document doi
+			const doi = getDocumentDoi(document);
+			if (!doi) return;
+
+			// get filename
+			const fileUrl = await getPDFURL(doi);
+			const filename = pdfNameFromUrl(fileUrl);
+
+			const filenames: string[] = [];
+			if (filename) filenames.push(filename);
+			// create document asset
+			const documentAsset: DocumentAsset = {
+				name,
+				description: name,
+				fileNames: filenames,
+				username
 			};
 
-			// FIXME: handle cases where assets is already added to the project
-
-			// first, insert into the proper table/collection
-			const res = await addDocuments(body);
-			if (res) {
-				const documentId = res.id;
-
-				// then, link and store in the project assets
-				const assetsType = AssetType.Publications;
-				await useProjects().addAsset(assetsType, documentId, projectId);
-			}
+			const newDocument: DocumentAsset | null = await createNewDocumentAsset(documentAsset);
+			if (!newDocument || !newDocument.id) return;
+			// upload document rom doi and attach it to the document asset if the pdf exists
+			await useProjects().addAsset(AssetType.Documents, newDocument.id, projectId);
+			if (!filename) return;
+			await addDocumentFromDOI(newDocument.id, doi, filename);
```

Comment:
This call should be done on the `hmi-server` after a successful PDF upload.

Reply:
@YohannParis I have changed this logic slightly.
So when there is no `filename` on a pdf it will not be able to upload at all.  I'm a little apprehensive on adding the adding the asset to the project in the backend just cause the endpoint might be doing too much, also it follows the pattern the ways we are adding models and datasets to the project (on the frontend).  let me know what you think.

Reply:
all logic has been added to backend